Directory.Build.props

```
@@ -1,31 +1,34 @@
 <Project>
-  <PropertyGroup>
-    <_ParentDirectoryBuildPropsPath Condition="'$(_DirectoryBuildPropsFile)' != ''">$([System.IO.Path]::Combine('..', '$(_DirectoryBuildPropsFile)'))</_ParentDirectoryBuildPropsPath>
-  </PropertyGroup>
+    <PropertyGroup>
+        <UseArtifactsOutput>true</UseArtifactsOutput>
+        <ArtifactsPath>$(MSBuildThisFileDirectory)artifacts</ArtifactsPath>
 
-  <Import Project="$(_ParentDirectoryBuildPropsPath)" Condition="Exists('$(_ParentDirectoryBuildPropsPath)')"/>
+        <LangVersion>latest</LangVersion>
+        <Features>strict</Features>
+        <DebugType>embedded</DebugType>
+        <Nullable>enable</Nullable>
+        <ImplicitUsings>enable</ImplicitUsings>
 
-  <PropertyGroup>
-    <WarningLevel>3</WarningLevel>
-    <AnalysisLevel>preview</AnalysisLevel>
-    <Features>strict</Features>
-  </PropertyGroup>
+        <EnableNETAnalyzers>true</EnableNETAnalyzers>
+        <AnalysisLevel>latest</AnalysisLevel>
+        <AnalysisMode>recommended</AnalysisMode>
+        <IsAotCompatible>true</IsAotCompatible>
+        <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
+        <ReportAnalyzer>true</ReportAnalyzer>
 
-  <PropertyGroup>
-    <ImplicitUsings>enable</ImplicitUsings>
-    <Nullable>enable</Nullable>
-  </PropertyGroup>
+        <ContinuousIntegrationBuild Condition="'$(GITHUB_ACTIONS)' == 'true'">true</ContinuousIntegrationBuild>
+        <!-- TODO re-enable -->
+        <!-- <TreatWarningsAsErrors Condition="'$(ContinuousIntegrationBuild)' == 'true' or '$(Configuration)' == 'Release'">true</TreatWarningsAsErrors> -->
```

C'mon now, nice try 🤓. Let's clean up all the warnings before we merge this, please re-enable. And let's make it unconditional. We shall always treat warnings as errors to help us avoid having warnings...


```suggestion
        <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
```

---

Most of the new warnings come from me setting `AnalysisMode` to `recommended`, I was hoping you would go over them and fix/disable ones you don't think are necessary.

> And let's make it unconditional

The idea is it should compile with warnings in Debug mode so it doesn't annoy you during development and prototyping.

---

What are some of the warnings with the recommended analysis mode? How much will need to change, will I find myself updating the `.editorconfig` more?